### DISCUSSION TITLE
Warning in SolidMechanics

### DISCUSSION THREAD

**Discussion by jin0123456** [idaholab/moose#28076] [Q&A Modules: Solid mechanics]
Hello

I met a warning,
```
*** Warning ***
Missing coupled variables {temp} (add them to coupled_variables parameter of TM_fuel0)
```
This warning occurs in `Physics/SolidMechanics/QuasiStatic`, but my MOOSE version is not the latest so it occurs in `Modules/TensorMechanics/Master`.
The solution is I need to add `temperature = temp`. But the convergence is poor when i did it. 
Since it's just warning the simulation can continue, should I necessarily add `temperature = temp`? And what's this setting about, would it make a big difference to the result?

**Comment by GiudGiud** [comment]
hello

Do you want to model thermal expansion or not? This is what it controls.
If convergence is poor with it, maybe the parameters for the thermal expansion are not set right. For example the reference temperature for the thermal eigenstrain. 

Guillaume

**Reply by jin0123456** [reply]
1.If i don't set `temperature = temp`, it still models thermal expansion. So whats the difference?
2.Besides, i set `stress_free_temperature = 300`, which 300 is the initial condition of temperature. 
Is that wrong?

**Reply by GiudGiud** [reply]
> 1.If i don't set temperature = temp, it still models thermal expansion. So whats the difference?

not sure but if you are getting a warning without then you should address it

What about the thermal expansion coefficient?
